gst/nnstreamer/nnstreamer_plugin_api_filter.h

```diff
@@ -28,23 +28,25 @@
 
 /** Macros for accelerator types */
 #define ACCL_NONE_STR "none"
+#define ACCL_DEF_STR  "default"
 #define ACCL_AUTO_STR "auto"
 #define ACCL_CPU_STR  "cpu"
 #define ACCL_GPU_STR  "gpu"
 #define ACCL_NPU_STR  "npu"
 #define ACCL_NEON_STR "neon"
 #define ACCL_SRCN_STR "srcn"
-#define ACCL_DEF_STR  "default"
+#define ACCL_TPU_STR  "tpu"
 
 #define REGEX_ACCL_EACH_ELEM(ELEM) "[^!]" ELEM "[,]?"
 
 #define REGEX_ACCL_AUTO REGEX_ACCL_EACH_ELEM(ACCL_AUTO_STR)
+#define REGEX_ACCL_DEF  REGEX_ACCL_EACH_ELEM(ACCL_DEF_STR)
 #define REGEX_ACCL_CPU  REGEX_ACCL_EACH_ELEM(ACCL_CPU_STR)
 #define REGEX_ACCL_GPU  REGEX_ACCL_EACH_ELEM(ACCL_GPU_STR)
 #define REGEX_ACCL_NPU  REGEX_ACCL_EACH_ELEM(ACCL_NPU_STR)
 #define REGEX_ACCL_NEON REGEX_ACCL_EACH_ELEM(ACCL_NEON_STR)
 #define REGEX_ACCL_SRCN REGEX_ACCL_EACH_ELEM(ACCL_SRCN_STR)
-#define REGEX_ACCL_DEF  REGEX_ACCL_EACH_ELEM(ACCL_DEF_STR)
+#define REGEX_ACCL_TPU  REGEX_ACCL_EACH_ELEM(ACCL_TPU_STR)
```

Comment:
We need to be more specific if we are going to designate an enum for each model of accelerator.

Reply:
Currently, these enums are generic ones. For example - ACCL_NPU denotes a generic NPU device. Which device will be used if chosen this accelerator (if multiple accelerators present) is dependent on the framework.
In the frameworks supported as extensions of tensor filter with nnstreamer, they do not give the option to choose the accelerator specific model out of the available ones.

Reply:
Embedding model specific information in these enums might be too much.
How about - model specific information can be taken in PROP_ACCELERATOR as input, parsed to be stored separately, and can be used later when frameworks support it?

Reply:
(This is suggestion, we'll need a lot of revises)
Keeping compatible with ml_nnfw_hw_e of nnstreamer.h, we may have parsing rules as:
```
cpu
gpu
npu (== Any NPU, 0x3000)
npu.sr (== Any SR NPU 0x3800)
npu.srasr (== SR-ASR-NPU 0x3801)
npu.srvision (== SR-Vision-NPU 0x3082)
npu.movidius (0x3001)
npu.edgetpu (== Google/Coral TPU, 0x3002)
npu.vivante (== Vivante NPU. 0x3003)
...
??

Reply:
And we need some explanations in the code; e.g., what is "SRCN"?

Reply:
Good suggestion - updated
Current implementation is not compatible with ml_nnfw_hw_e - 
NPU enum value is 3000 in ml_nnfw_hw_e and is 4000 internally (to avoid bit overlap) - should the enum value be updated to 3000 internally as well?